x-pack/plugins/lists/server/services/extension_points/extension_point_storage_client.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from 'kibana/server';
+
+import { ExtensionPoint, NarrowExtensionPointToType } from './types';
+import { ExtensionPointError } from './errors';
+import { ExtensionPointStorageInterface } from './extension_point_storage';
+
+export class ExtensionPointStorageClient {
+  constructor(
+    private readonly storage: ExtensionPointStorageInterface,
+    private readonly logger?: Logger
+  ) {}
+
+  /**
+   * Retrieve a list (`Set`) of extension points that are registered for a given type
+   * @param extensionType
+   */
+  get<T extends ExtensionPoint['type']>(
+    extensionType: T
+  ): Set<NarrowExtensionPointToType<T>> | undefined {
+    return this.storage.get(extensionType);
+  }
+
+  /**
+   * Runs a set of callbacks by piping the Response from one extension point callback to the next callback
+   * and finally returning the last callback payload.
+   *
+   * @param extensionType
+   * @param initialCallbackInput The initial argument given to the first extension point callback
+   * @param callbackResponseValidator A function to validate the returned data from an extension point callback
+   */
+  async pipeRun<
+    T extends ExtensionPoint['type'],
+    D extends NarrowExtensionPointToType<T> = NarrowExtensionPointToType<T>,
+    P extends Parameters<D['callback']> = Parameters<D['callback']>
+  >(
+    extensionType: T,
+    initialCallbackInput: P[0],
+    callbackResponseValidator?: (data: P[0]) => Error | undefined
+  ): Promise<P[0]> {
+    let inputArgument: P[0] = initialCallbackInput;
```

Comment:
nit: You can use `let inputArgument = initialCallbackInput;`, you don't have to retype it again here.